### PR TITLE
Fix/Feature/class-refactor

### LINKED ISSUE #4: Sleep Data
- [x] For a user (identified by their userID), the average number of hours slept per day
- [x] For a user, their average sleep quality per day over all time
- [x] For a user, how many hours they slept for a specific day (identified by a date)
- [x] For a user, their sleep quality for a specific day (identified by a date)
- [x] For a user, how many hours slept each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] For a user, their sleep quality each day over the course of a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] For all users, the average sleep quality
- [x] Find all users who average a sleep quality greater than 3 for a given week (7 days) - you should be able to calculate this for any week, not just the latest week
- [x] For a given day (identified by the date), find the users who slept the most number of hours (one or more if they tied)

### LINKED ISSUE #31: Add Parent Class (Calculation)
Calculation class should be a parent class

- [x] -Activity, Hydration, Sleep should inherit it's methods and properties 
- [x] -Test Methods

### PR DESCRIPTION
**Pull request by alexthompson207**
### Description
* Is this a feature or a fix?

This is both a fix and and a feature.  
* Where should the reviewer start?

Start in the `Calculation.js` class file and its respective test. Then look at `Sleep`, `Hydration`, and `Activity` classes and their tests. Then look at `User` and `UserRepo` classes and their tests.

* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?

Added a parent class called `Calculation` and its respective test. Created 3 methods to calculate Average, DailyData and Weekly data. These changes were made because in `Sleep`, `Hydration`, and `Activity` classes some of their methods were reused across each class. Also added nested `describe` blocks for each test in every class to break up tests on properties and methods.

* What outside features or research did you use to implement this fix?

Just lesson from class.

* How should this be tested?

run `npm test` in console. All 70 tests should pass.

### Relevant Screen Shots
***
#### Please review considerations below:
- [ ] Follows Turing Style Guidelines
- [ ] Changes generate no new warnings or errors
- [ ] Checked code for and corrected any spelling errors
- [ ] README has been updated(if applicable)

### CODE REVIEW

File: test/userrepo-test.js [hunk 1]
```diff
@@ -222,14 +228,14 @@ describe('User Repo', () => {
...
         '2': [4.7],
         '3': [3]
       });
-      expect(userRepo.isolateUsernameAndRelevantData( 'numOunces', weeklyHydrationData)).to.eql({
-        '1': [ 37 ],
-        '2': [ 38 ] 
+      expect(userRepo.isolateUsernameAndRelevantData('numOunces', weeklyHydrationData)).to.eql({
+        '1': [37],
+        '2': [38]
       });
```
Comment (aemiers) [inline test/userrepo-test.js:239]: Excellent job with the Calculations class! Maybe next time break it up into different branches?

Comment (alexthompson207) [reply test/userrepo-test.js:239]: Yes I totally agree! It started as a refactor branch but then took a turn into a feature with a new class and also refactoring. Next time!

File: src/Activity.js [hunk 2]
```diff
@@ -102,6 +90,4 @@ class Activity {
   }
 }
 
-
-
-export default Activity;
+export default Activity;
```
Comment (itsnameissteven) [inline src/Activity.js:93]: This looks way cleaner now

**Review by itsnameissteven — APPROVED** [review]
Wow that was a lot of work. Good job on this. I think it'll make it easier on us moving forward having a cleaner code base

File: src/Activity.js (hunk 2 above)
Comment (alexthompson207) [reply src/Activity.js:93]: Awesome, thanks. We might even be able to remove a few more methods that find information that the spec doesn't ask for!

**Pull request merged**